### ISSUE TITLE
[API] Drawer->text takes AbstractFont as parameter

### ISSUE THREAD

**Issue by rpiotaix** [avalanche123/Imagine#493]
Shouldn't it be taking a FontInterface instead?

My code looks like this:

```php
$text = 'blah';
$font = $this->imagine->font($this->fontPath, 12, $image->palette()->color('#FFFFFF'));
$image->draw()->text($text, $font, new Point(1,1));
```
But PhpStorm complains about the type of $font (expects AbstractFont, got FontInterface). Am I doing sth wrong?

**Comment by vinikey** [comment]
Hello!

Is it possible to make 2-3 examples about the use of draw()->text, cause is really difficult to understand the doc page (https://imagine.readthedocs.io/en/stable/usage/drawing.html#text) and what do you mean with the Font class, since we have only AbstractFont and FontInterface classes.

Text on images is one of the most important tasks on image manipulation and any help on this would be so appreciated!

Thank you!

**Comment by mlocati** [comment]
> we have only AbstractFont and FontInterface classes.

Nope: we also have the `Imagine\Gd\Font`, `Imagine\Imagick\Font`, and `Imagine\Gmagick\Font` classes.

In order to see some examples, you can look at the tests ([this one for example](https://github.com/avalanche123/Imagine/blob/1.2.2/tests/tests/Draw/AbstractDrawerTest.php#L220)).
